lib/jxl/encode.cc

```diff
@@ -130,11 +133,12 @@ JxlEncoderStatus JxlEncoderStruct::RefillOutputByteQueue() {
 JxlEncoderStatus JxlEncoderSetColorEncoding(JxlEncoder* enc,
                                             const JxlColorEncoding* color) {
   if (enc->color_encoding_set) {
-    // Already set
-    return JXL_ENC_ERROR;
+    enc->error = JXL_ENC_ERR_API_USAGE;
+    return JXL_API_ERROR("Color encoding is already set");
```

Comment:
In brotli we had a macro/function that sets the enc->error and returns the JXL_ENC_ERROR value. Using such a macro (extending JXL_API_ERROR) instead of manually setting enc->error before returning  would avoid missing places where we should have set enc->error but we forgot. Something like `#define JXL_API_ERROR(enc, error_code, message) ....`

Reply:
Updated.